src/test/java/com/patina/codebloom/db/achievements/AchievementRepositoryTest.java

```diff
@@ -0,0 +1,120 @@
+package com.patina.codebloom.db.achievements;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import com.patina.codebloom.common.db.models.achievements.Achievement;
+import com.patina.codebloom.common.db.repos.achievements.AchievementSqlRepository;
+
+@SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(OrderAnnotation.class)
+public class AchievementRepositoryTest {
+
+    private AchievementSqlRepository repo;
+    private Achievement testAchievement;
+    private Achievement deletableAchievement;
+    private String mockUserId = "ed3bfe18-e42a-467f-b4fa-07e8da4d2555";
+
+    @Autowired
+    public AchievementRepositoryTest(final AchievementSqlRepository repo) {
+        this.repo = repo;
+    }
+
+    @BeforeAll
+    void createAchievement() {
+        testAchievement = Achievement.builder()
+                        .userId(mockUserId)
+                        .iconUrl("")
+                        .title("Test Achievement")
+                        .description("Integration test achievement")
+                        .isActive(true)
+                        .createdAt(OffsetDateTime.now())
+                        .deletedAt(null)
+                        .build();
+
+        repo.createAchievement(testAchievement);
+    }
+
+    @AfterAll
+    void cleanUp() {
+        boolean isSuccessful = repo.deleteAchievementById(testAchievement.getId());
+        if (!isSuccessful) {
+            fail("Failed deleting achievement by id.");
+        }
+    }
+
+    @Test
+    @Order(1)
+    void testGetAchievementById() {
+        Achievement found = repo.getAchievementById(testAchievement.getId());
+        assertNotNull(found);
+        assertEquals(testAchievement.getId(), found.getId());
+    }
```

Comment:
test the equals directly as well, so we can ensure we don't lose any data in the round trip from server to db, and back.

```suggestion
        assertEquals(testAchievement.getId(), found.getId());
        assertEquals(testAchievement, found);
    }
```